File: TopCoder05/Ruby/Main.rb
class ThePalindrome
  def find(s)
    i = s.size
    while true
      fleg = true
      for j in 0...s.size
        if (i - j - 1) < s.size && s[j] != s[i - j - 1]
          fleg = false
          break
        end
      end
      if fleg
        return i
      end
      i += 1
    end
  end
end

s = "qwerty"
puts ThePalindrome.new.find(s)

File: TopCoder05/PHP/Main.php
<?php
	class ThePalindrome {
		public function find ($s){
			for($i = strlen($s);; $i++){
				$fleg = true;
				for($j = 0 ; $j < strlen($s); $j++){
					if(($i - $j - 1) < strlen($s) && $s{$j} != $s{$i - $j -1}){
						$fleg = false;
						break;
					}
				}
				if($fleg){
					return $i;
				}
				
			}
		}
	}
	
	$s = "qwerty";
	$tp = new ThePalindrome;
	echo $tp->find($s);

?>